services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -966,6 +981,43 @@ public Response testLDAPConnection(TestLdapConnectionRepresentation config) {
                 config.getStartTls());
     }
 
+    /**
+     * Get LDAP supported extensions.
+     * @param config LDAP configuration
+     * @return
+     */
+    @POST
+    @Path("ldap-supported-extensions")
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(javax.ws.rs.core.MediaType.APPLICATION_JSON)
+    public Response getLDAPSupportedExtensions(TestLdapConnectionRepresentation config) {
+
+        auth.realm().requireManageRealm();
+
+        String bindCredential = handleSavedBindCredential(config.getComponentId(), config.getBindCredential());
+
+        // Create Factory Methods for LDAPConfig and LDAPIdentityStore in LDAPConnectionTestManager?
+        MultivaluedHashMap<String, String> map = new MultivaluedHashMap<>();
+        map.add(LDAPConstants.CONNECTION_URL, config.getConnectionUrl());
+        map.add(LDAPConstants.USE_TRUSTSTORE_SPI, config.getUseTruststoreSpi());
+        map.add(LDAPConstants.AUTH_TYPE, LDAPConstants.AUTH_TYPE_SIMPLE);
+        map.add(LDAPConstants.BIND_DN, config.getBindDn());
+        map.add(LDAPConstants.BIND_CREDENTIAL, bindCredential);
+        IdentityStore ldapIdentityStore = new LDAPIdentityStore(session, new LDAPConfig(map));
+
+        Set<LDAPOid> ldapOids = ldapIdentityStore.queryServerCapabilities();
```

Comment:
It will be better if response allows to differentiate between supportedControl, supportedExtensions and supportedFeatures. The JSON can possibly look like this:
```
{ 
    "supportedExtension": [ "1.3.6.1.4.1.1466.20036" ],
    "supportedFeatures": [ "1.3.6.1.1.14", "1.3.6.1.4.1.4203.1.5.1" ],
    "supportedControl": [ "1.2.840.113556.1.4.1413", "1.2.840.113556.1.4.2239", "1.2.840.113556.1.4.319" ]
}
````
Feel free to create some internal JSON representation class for help with this - for example see TestLdapConnectionRepresentation for the inspiration

Reply:
It will be also good to dump those supported extensions, features and controls to the server.log with the "log.info"

Reply:
I though about that too, but my conclusion was it is practically not relevant whether a given OID is a _control_, _extension_ or _feature_. That is way I refactored the method to "capabilities" in the first place. Do you think it is really necessary? The OID is unique anyway. Good point with the info log. Will be added.

Reply:
I think that it may be useful in general to differentiate between control, extension and feature. 

LDAP server itself also differentiates between those 3 things, so from this perspective, differentiation makes sense :)

Reply:
Done.